features/FEATURE_BLE/ble/SecurityManager.h

```diff
@@ -74,22 +183,190 @@ class SecurityManager {
+
+        ////////////////////////////////////////////////////////////////////////////
+        // Pairing
+        //
+
+        /**
+         * Request application to accept or reject pairing. Application should respond by
+         * calling the appropriate function: acceptPairingRequest or cancelPairingRequest
+         *
+         * @param[in] connectionHandle connection connectionHandle
+         */
+        virtual void pairingRequest(ble::connection_handle_t connectionHandle) {
+            (void)connectionHandle;
+        }
+
+        /**
+         * Indicate to the application that pairing has completed.
+         *
+         * @param[in] connectionHandle connection connectionHandle
+         * @param[in] result result of the pairing indicating success or reason for failure
+         */
+        virtual void pairingResult(ble::connection_handle_t connectionHandle, SecurityCompletionStatus_t result) {
+            (void)connectionHandle;
+            (void)result;
+        }
+
+        ////////////////////////////////////////////////////////////////////////////
+        // Security
+        //
+
+        /**
+         * Deliver the requested whitelist to the application.
+         *
+         * @param[in] whitelist pointer to the whitelist filled with entries based on bonding information
+         */
+        virtual void whitelistFromBondTable(Gap::Whitelist_t* whitelist) {
+            if (whitelist) {
+                delete whitelist;
+            }
+        }
+
+        ////////////////////////////////////////////////////////////////////////////
+        // Encryption
+        //
+
+        /**
+         * Inform the device of the encryption state of a given link.
+         *
+         * @param[in] connectionHandle connection connectionHandle
+         * @param[in] result encryption state of the link
+         */
+        virtual void linkEncryptionResult(ble::connection_handle_t connectionHandle, ble::link_encryption_t result) {
+            (void)connectionHandle;
+            (void)result;
+        }
+
+        ////////////////////////////////////////////////////////////////////////////
+        // MITM
+        //
+
+        /**
+         * Display the given passkey on the local device.
+         *
+         * @param[in] connectionHandle connection connectionHandle
+         * @param[in] passkey 6 digit passkey to be displayed
+         */
+        virtual void passkeyDisplay(ble::connection_handle_t connectionHandle, const SecurityManager::Passkey_t passkey) {
+            (void)connectionHandle;
+            (void)passkey;
+        }
+
+        /**
+         * Indicate to the application that a confirmation is required. The application should
+         * proceed by supplying the confirmation confirmationEntered function.
+         *
+         * @param[in] connectionHandle connection connectionHandle
+         */
+        virtual void confirmationRequest(ble::connection_handle_t connectionHandle) {
```

Comment:
What are you supposed to confirm ? I believe this should take the value to confirm in parameter.

Reply:
those comments are bad, will clarify, this is confirmation that passkeys match

Reply:
How do you push the confirmation value to the user ? According to the specification: 

```
If both devices have output capabilities, this step requires the displaying of a
user confirmation value. This value should be displayed until the end of step 2.
If one or both devices do not have output capabilities, the same protocol is
used but the Hosts will skip the step asking for the user confirmation.
```

Reply:
you use displayPasskey on both (supplying the number to display) and then request confirmation on one or both and application calls confirmationEntered

Reply:
Yes you're right!

Reply:
added diagram to overview doc and comments in doxygen